prospero-cli/src/main/java/org/wildfly/prospero/cli/CliMessages.java

```diff
@@ -638,18 +646,40 @@ default String serverVersionsHeader() {
         return bundle.getString("prospero.channels.versions.header");
     }
 
-    default  IllegalArgumentException unknownStabilityLevel(String stabilityLevel, List<String> supportedStabilityLevels) {
+    default IllegalArgumentException unknownStabilityLevel(String stabilityLevel, List<String> supportedStabilityLevels) {
         return new IllegalArgumentException(format(bundle.getString("prospero.install.unknown_stability_level"),
                 stabilityLevel, String.join(",", supportedStabilityLevels)));
     }
+
     default String featurePackRequiresLayers(String featurePackName) {
         return format(bundle.getString("prospero.features.add.validation.layers_required"), featurePackName, CliConstants.LAYERS);
     }
+
     default String featurePackDoesNotSupportCustomization(String featurePackName) {
         return format(bundle.getString("prospero.features.add.validation.customization_not_supported"), featurePackName,
                 String.join(",", List.of(CliConstants.LAYERS, CliConstants.TARGET_CONFIG)));
     }
+
     default String featurePackRequiresLicense(String featurePackName) {
         return format(bundle.getString("prospero.features.add.required_licences"), featurePackName);
     }
+
+    default IllegalArgumentException missingRequiredParameter(String param) {
```

Comment:
Can you change this to throw `CommandLine.ParameterException`. I found out this way picocli will print the full syntax/help message. (https://picocli.info/#_custom_validation)